Updates

MikaelSmith opened the pull request:
##### (maint) Do make install on Windows

```
Test `make install` on Windows. Also configure install location on
Travis via CMake rather than make's DESTDIR (behavior is slightly
different).
```
##### (maint) Allow configuration of static/shared library

```
Consumers building the project should be able to determine whether they
want a static or shared library. Don't hard-code the option, so it can
be configured externally.
```
##### (maint) Let CMake determine correct binary location

```
CMake already knows how to differentiate between runtime, library, and
archive files on Windows. Use the correct install options to do that
rather than hard-coding that dynamic libraries should be installed to
bin (including their import libs, which should be in lib).
```
##### (maint) Bump Leatherman
##### (maint) Use Boost 1.58

```
Internal builds were moved to Boost 1.58, use it for AppVeyor as well.
```
##### (maint) Use visibility on all platforms except Mac

Hides symbols by default and requires explicitly exporting the interface
on all platforms except Apple. On Apple this introduced issues in
Facter, though I don't recall the reason.

Commit:
(maint) Use Boost 1.58

Internal builds were moved to Boost 1.58, use it for AppVeyor as well.

Commit:
(maint) Bump Leatherman

Commit:
(maint) Let CMake determine correct binary location

CMake already knows how to differentiate between runtime, library, and
archive files on Windows. Use the correct install options to do that
rather than hard-coding that dynamic libraries should be installed to
bin (including their import libs, which should be in lib).

Commit:
(maint) Allow configuration of static/shared library

Consumers building the project should be able to determine whether they
want a static or shared library. Don't hard-code the option, so it can
be configured externally.

Commit:
(maint) Do make install on Windows

Test `make install` on Windows. Also configure install location on
Travis via CMake rather than make's DESTDIR (behavior is slightly
different).

Commit:
(maint) Use visibility on all platforms except Mac

Hides symbols by default and requires explicitly exporting the interface
on all platforms except Apple. On Apple this introduced issues in
Facter, though I don't recall the reason.

branan commented on lib/CMakeLists.txt:
```diff
@@ -37,15 +37,18 @@ target_link_libraries(lib${PROJECT_NAME} PRIVATE
-    add_compiler_export_flags()
-elseif (WIN32)
+if (NOT APPLE)
     add_compiler_export_flags()
```
Does this need to be skipped on AIX as well?

MikaelSmith replied:
On AIX it will skip because it's not supported.

On Apple it's supported, but behaves in weird ways. I think mostly it gives warnings about boost, and I don't want to rebuild boost yet.

branan replied:
OK

branan commented:
:+1:

Commit:
(maint) Update cmake URL but ignore certificate

cmake.org updated to redirect to HTTPS and use a shorter URL.
However the certificate they use requires validation from an SSL
chain that's not installed on Ubuntu 12.04. Since we were previously
using HTTP, just ignore the certificate for now while this is
resolved.

Commit:
(maint) Generalize Git fixup to remove sh.exe from path

CMake won't produce mingw makefiles with sh.exe in the PATH, so we
ensure it isn't. The previous method only worked with the 32-bit version
of Git. Generalize to work with both 32-bit and 64-bit versions of Git.
